IOS/SwiftSeedProject/SwiftSeedProject/Application/Service/BaseService.swift

```diff
@@ -7,25 +7,34 @@
 //
 
 import Foundation
+import SwiftyJSON
 
 class BaseService<T: PersistenceObject>  {
     
-    private let persistence: Persistence
-    private let restClient: AnyObject? = nil
-    
-    //TODO: we should inclued the API access also
-    init(persistence: Persistence) {
+    private let persistence: Persistence!
+    internal let api: Api!
+
+    init(persistence: Persistence, api: Api) {
         self.persistence = persistence
+        self.api = api
     }
     
     public func getAll() -> [T] {
         return persistence.getAll(entityName: T.EntityName)
     }
     
-    public func getEntityBy(identifier: Int) -> T {
+    public func getEntityBy(identifier: String) -> T? {
         return persistence.getBy(entityIdentifier: identifier, entityName: T.EntityName)
     }
     
+    public func createEntity(json: JSON) {
```

Comment:
I don't think the network layer should persist objects, because it would be mixing responsibilities and also creating a new dependency on persistance on the Network layer. The service is already delegating the save to the Persistence. What do you think? Maybe I'm not getting something